contracts/OgvStaking.sol

```diff
@@ -222,45 +246,40 @@ contract OgvStaking is ERC20Votes {
     /// @param user to preview rewards for
     /// @return OGV rewards amount
     function previewRewards(address user) external view returns (uint256) {
-        uint256 supply = totalSupply();
-        if (supply == 0) {
+        if (totalSupply() == 0) {
             return 0; // No one has any points to even get rewards
         }
-        uint256 _accRewardPerShare = accRewardPerShare;
-        _accRewardPerShare += (rewardsSource.previewRewards() * 1e12) / supply;
-        uint256 netRewardsPerShare = _accRewardPerShare - rewardDebtPerShare[user];
+        uint256 netRewardsPerShare = accRewardPerShare - rewardDebtPerShare[user];
         return (balanceOf(user) * netRewardsPerShare) / 1e12;
     }
 
     /// @dev Internal function to handle rewards accounting.
     ///
-    /// 1. Collect new rewards for everyone
-    /// 2. Calculate this user's rewards and accounting
-    /// 3. Distribute this user's rewards
+    /// 1. Calculate this user's rewards and accounting
+    /// 2. Distribute this user's rewards, if any
     ///
     /// This function *must* be called before any user balance changes.
     ///
     /// This will always update the user's rewardDebtPerShare to match
-    /// accRewardPerShare, which is essential to the accounting.
+    /// accRewardPerShare, which is essential to the accounting. This
+    /// wouldn't allow user to claim rewards twice
     ///
     /// @param user to collect rewards for
-    function _collectRewards(address user) internal {
-        uint256 supply = totalSupply();
-        if (supply > 0) {
-            uint256 preBalance = ogv.balanceOf(address(this));
-            try rewardsSource.collectRewards() {}
-            catch {
-                // Governance staking should continue, even if rewards fail
-            }
-            uint256 collected = ogv.balanceOf(address(this)) - preBalance;
-            accRewardPerShare += (collected * 1e12) / supply;
+    /// @param netRewards Net reward collected for user
+    function _collectRewards(address user) internal returns (uint256 netRewards) {
+        if (totalSupply() == 0) {
+            return 0; // No one has any points to even get rewards
```

Comment:
I guess because points are both collected before unstaking, and new sakes are disabled, the system can never have zero points and still have undistributed rewards, so this is correct. Does look the wrong change at first glance though because the previous 0 > guard is for on collecting points into the contract's accounting, not on distributing points which the new if statement is guarding.